examples/C++/DDS/BasicConfigurationExample/BasicConfigurationSubscriber.cpp

```diff
@@ -0,0 +1,252 @@
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    // REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    // CREATE THE SUBSCRIBER
+    subscriber_ = participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT, nullptr);
+
+    if (subscriber_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE TOPIC
+    topic_ = participant_->create_topic(
+        topic_name,
+        "HelloWorld",
+        TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE READER
+    if (max_messages > 0)
+    {
+        listener_.set_max_messages(max_messages);
+    }
+    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+    if (!transport.empty())
+    {
+        rqos.data_sharing().off();
+    }
+    else
+    {
+        rqos.data_sharing().automatic();    // default
+    }
+
+    if (reliable)
+    {
+        rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+    }
+    else
+    {
+        rqos.reliability().kind = BEST_EFFORT_RELIABILITY_QOS;  // default
+    }
+
+    if (transient)
+    {
+        rqos.durability().kind = TRANSIENT_LOCAL_DURABILITY_QOS;
+    }
+    else
+    {
+        rqos.durability().kind = VOLATILE_DURABILITY_QOS;   // default
+    }
+
+    reader_ = subscriber_->create_datareader(topic_, rqos, &listener_);
+
+    if (reader_ == nullptr)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+HelloWorldSubscriber::~HelloWorldSubscriber()
+{
+    if (participant_ != nullptr)
+    {
+        if (topic_ != nullptr)
+        {
+            participant_->delete_topic(topic_);
+        }
+        if (subscriber_ != nullptr)
+        {
+            if (reader_ != nullptr)
+            {
+                subscriber_->delete_datareader(reader_);
+            }
+            participant_->delete_subscriber(subscriber_);
+        }
+        DomainParticipantFactory::get_instance()->delete_participant(participant_);
+    }
+}
+
+void HelloWorldSubscriber::SubListener::set_max_messages(
+        uint32_t max_messages)
+{
+    max_messages_ = max_messages;
+}
+
+void HelloWorldSubscriber::SubListener::on_subscription_matched(
+        DataReader*,
+        const SubscriptionMatchedStatus& info)
+{
+    if (info.current_count_change == 1)
+    {
+        matched_ = info.current_count;
+        std::cout << "Subscriber matched." << std::endl;
+    }
+    else if (info.current_count_change == -1)
+    {
+        matched_ = info.current_count;
+        std::cout << "Subscriber unmatched." << std::endl;
+    }
+    else
+    {
+        std::cout << info.current_count_change
+                  << " is not a valid value for SubscriptionMatchedStatus current count change" << std::endl;
+    }
+}
+
+void HelloWorldSubscriber::SubListener::on_data_available(
+        DataReader* reader)
+{
+    SampleInfo info;
+    while (reader->take_next_sample(&hello_, &info) == ReturnCode_t::RETCODE_OK)
```

Comment:
This should also stop when the Subscriber stops

Reply:
:+1: